intel_extension_for_transformers/backends/neural_engine/graph/jblas/jblas/jit_blas_weight_compression.h

```diff
@@ -1105,63 +1217,78 @@ namespace gemm_default {
-        jblas::epilogue::gemm::AccumulatorWriteBack<float, float>>,
+        jblas::prologue::gemm::ActivationF32U8KBlockQuantize,
+        jblas::prologue::weight_comp::gemm_kblcok::WeightS4ClipScaleFp32,
+        jblas::epilogue::gemm::AccumulatorWriteBackFp32>,
     jblas::utils::parallel::Parallel2DGemmKBlockFixed>;
 
 }  // namespace avx512_vnni
 namespace amx_bf16 {
 JBLAS_ISA constexpr DefaultISA = JblasAMX_BF16;
-using GemmKernelS4KBlock = jblas::wrapper::gemm_pack_weight::GemmInterfacePackWeight<
+using GemmKernelS4FullRangeFp32KBlock = jblas::wrapper::gemm_pack_weight::GemmInterfacePackWeight<
     jblas::wrapper::gemm_pack_weight::GemmLauncherPackWeight<
         DefaultISA, jblas::gemm::GemmCore_Row_NN_16x64_AMX_BF16,
         jblas::prologue::gemm::ActivationConverterFp32,  // activation fp32->bf16
-        jblas::prologue::weight_comp::gemm::WeightS4_KBlock,
-        jblas::epilogue::gemm::AccumulatorWriteBack<float, float>>,  // output fp32->fp32
+        jblas::prologue::weight_comp::gemm_kblcok::WeightS4FullRangeScaleFp32,
+        jblas::epilogue::gemm::AccumulatorWriteBackFp32>,  // output fp32->fp32
     DefaultParallel>;
-using GemmKernelFp4KBlock = jblas::wrapper::gemm_pack_weight::GemmInterfacePackWeight<
+using GemmKernelS4ClipFp32KBlock = jblas::wrapper::gemm_pack_weight::GemmInterfacePackWeight<
     jblas::wrapper::gemm_pack_weight::GemmLauncherPackWeight<
         DefaultISA, jblas::gemm::GemmCore_Row_NN_16x64_AMX_BF16,
         jblas::prologue::gemm::ActivationConverterFp32,  // activation fp32->bf16
-        jblas::prologue::weight_comp::gemm::WeightFp4_KBlock,
-        jblas::epilogue::gemm::AccumulatorWriteBack<float, float>>,  // output fp32->fp32
+        jblas::prologue::weight_comp::gemm_kblcok::WeightS4ClipScaleFp32,
+        jblas::epilogue::gemm::AccumulatorWriteBackFp32>,  // output fp32->fp32
+    DefaultParallel>;
+using GemmKernelFp4KBlock = jblas::wrapper::gemm_pack_weight::GemmInterfacePackWeight<
+    jblas::wrapper::gemm_pack_weight::GemmLauncherPackWeight<
+        DefaultISA, jblas::gemm::GemmCore_Row_NN_16x64_AMX_BF16,  // MXNXK = 16x64x32
+        jblas::prologue::gemm::ActivationConverterFp32,           // activation fp32->bf16
+        jblas::prologue::weight_comp::gemm_kblcok::WeightFp4BnbScaleFp32,
+        jblas::epilogue::gemm::AccumulatorWriteBackFp32>,  // output fp32->fp32
     DefaultParallel>;
 }  // namespace amx_bf16
 namespace amx_int8 {
 JBLAS_ISA constexpr DefaultISA = JblasAMX_INT8;
-using GemmSKernelDynamicS4KBlock = jblas::wrapper::gemm_kblock::GemmInterfaceKBlockPackWeight<
+using GemmSKernelDynamicS4ClipFp32KBlock = jblas::wrapper::gemm_kblock::GemmInterfaceKBlockPackWeight<
+    jblas::wrapper::gemm_kblock::GemmSLauncherKBlockPackWeight<
+        DefaultISA, jblas::gemm::kblock::GemmCore_Row_NN_16x48_AMX_INT8_KBLOCK,
+        jblas::prologue::gemm::ActivationF32S8KBlockQuantize,
```

Comment:
seems that we don't update the `prologueA` template with `ActivationFp32SymS8Quantize`, will we replace these old templates like `ActivationF32S8KBlockQuantize` in weight_compression in the future? can `ActivationFp32SymS8Quantize` support dynamic_quant which amx_int8/vnni gemmcore needed?

Reply:
ActivationF32S8KBlockQuantize is the KBLOCK quantize template, it binds with KBLOCK GemmCore. ActivationFp32SymS8Quantize is the per-channel quantize template, it binds with non-KBLOCK GemmCore.